Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift

```diff
@@ -82,25 +82,16 @@ extension String {
         }
         return sum % 10 == 0
     }
-
-    func decodeClientTokenBase64() -> DecodedClientToken {
-        let bytes = self.components(separatedBy: ".")
-        for element in bytes {
-            // decode element, add necessary padding to base64 to ensure it's a multiple of 4 (required by Swift foundation)
-            if let decodedData = Data(base64Encoded: element.padding(toLength: ((element.count + 3) / 4) * 4, withPad: "=", startingAt: 0)) {
-                let decodedString = String(data: decodedData, encoding: .utf8)!
-                if decodedString.contains("\"accessToken\":") {
-                    do {
-                        let token = try JSONDecoder().decode(DecodedClientToken.self, from: decodedData)
-                        return token
-                    } catch {
-                        log(logLevel: .error, title: "PARSER", message: "Failed to parse client token", prefix: nil, suffix: nil, bundle: Bundle.primerFrameworkIdentifier, file: #file, className: nil, function: #function, line: #line)
-                    }
-                }
-            }
-
-        }
-        return DecodedClientToken()
+    
+    var decodedJWTToken: DecodedClientToken? {
```

Comment:
is this tested?

Reply:
Yeah, quite a lot actually. It's one of the snippets that I was using on my previous company. I'd need to write some test cases though. I do think however that I should change the name, cause this isn't the whole token information, just the payload. @xevious78 @ceriksson would we need any other information from the token? Maybe I should ask Will about it